lib/model/model.go

```diff
@@ -306,29 +304,22 @@ func (m *Model) Completion(device protocol.DeviceID, folder string) float64 {
 		return 0 // Folder doesn't exist, so we hardly have any of it
 	}
 
-	rf.WithGlobalTruncated(func(f db.FileIntf) bool {
-		if !f.IsDeleted() {
-			tot += f.Size()
-		}
-		return true
-	})
-
+	_, _, tot := rf.GlobalSize()
 	if tot == 0 {
 		return 100 // Folder is empty, so we have all of it
 	}
 
 	var need int64
 	rf.WithNeedTruncated(device, func(f db.FileIntf) bool {
-		if !f.IsDeleted() {
-			need += f.Size()
-		}
+		need += f.Size()
```

Comment:
Yeah, it's what I'll look into next. Its probably similar to tracking the global stuff, except that changing the global can affect the need list for every device in the list. Still, better to more work at that stage than having to walk everything to figure out the need size.

Possibly we could even track the needed files themselves, so we don't have to walk global to get the need list. I'll look into that, but not in this PR.